src/Kernel/IQSharpEngine.cs

```diff
@@ -75,9 +79,71 @@ IMagicSymbolResolver magicSymbolResolver
                 Process.GetCurrentProcess().Id
             );
 
+
             eventService?.TriggerServiceInitialized<IExecutionEngine>(this);
         }
 
+        /// <summary>
+        ///     Registers an event handler that searches newly loaded packages
+        ///     for extensions to this engine (in particular, for result encoders).
+        /// </summary>
+        private void RegisterPackageLoadedEvent(IServiceProvider services, ILogger logger, IReferences references)
+        {
+            var knownAssemblies = references
+                .Assemblies
+                .Select(asm => asm.Assembly.GetName())
+                .ToImmutableHashSet()
+                // Except assemblies known at compile-time as well.
+                .Add(typeof(StateVectorToHtmlResultEncoder).Assembly.GetName());
+            foreach (var knownAssembly in knownAssemblies) System.Console.WriteLine($"{knownAssembly.FullName}");
+
+            // Register new display encoders when packages load.
+            references.PackageLoaded += (sender, args) =>
+            {
+                logger.LogDebug("Scanning for display encoders after loading {Package}.", args.PackageId);
+                foreach (var assembly in references.Assemblies.Select(asm => asm.Assembly))
+                {
+                    if (!knownAssemblies.Contains(assembly.GetName()))
```

Comment:
[nit] It looks like you could include this if condition as a `.Where()` clause in the `foreach` statement above.

Reply:
Good point, agreed.